src/coreclr/classlibnative/bcltype/objectnative.cpp

```diff
@@ -99,32 +127,30 @@ FCIMPL1(INT32, ObjectNative::GetHashCode, Object* obj) {
 
     OBJECTREF objRef(obj);
 
-    {
-        DWORD bits = objRef->GetHeader()->GetBits();
+    INT32 ret = TryGetHashCodeHelper(objRef);
```

Comment:
This is likely going to regress `ObjectNative::GetHashCode` microbenchmarks. There is going to be an extra conditional jump. I am not sure about the best way to avoid it. It may be best to just duplicate the implementation.

Reply:
`GetHashCode` could be a part of fairly hot code. I think trading code for perf (i.e. duplicating something) would be acceptable.

Reply:
I reverted the change to the existing method in favor of duplicating a small amount of code.